File: frontend/lib/models/patient.dart
import 'package:frontend/models/user.dart';

import '../utility.dart';

class Patient extends User {
  String? symptoms;

  Patient({userID, firstName, lastName, email, dateOfBirth, accountStatus, required this.symptoms})
      : super(
            userID: userID,
            firstName: firstName,
            lastName: lastName,
            email: email,
            dateOfBirth: dateOfBirth,
            accountStatus: accountStatus);

  factory Patient.fromJson(Map<String, dynamic> json) {
    return Patient(
      userID: json['userID'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      dateOfBirth: DateTime.parse(json['dateOfBirth']),
      accountStatus: json['accountStatus'],
      symptoms: json['symptoms'],
    );
  }

  toJson(String? newPassword) {
    return {
      'userID': userID,
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'password': newPassword,
      'dateOfBirth': Utility.dateToStringJSON(dateOfBirth),
      'accountStatus': accountStatus,
      'symptoms': symptoms,
    };
  }

}

File: backend/auth-microservice/src/main/java/sept/superfive/authmicroservice/model/Patient.java
package sept.superfive.authmicroservice.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Patient extends User {
    private String symptoms;

    public Patient(int userID, String firstName, String lastName, String email, String password,
                   LocalDate dateOfBirth, List<Role> roles) {
        super(userID, firstName, lastName, email, password, dateOfBirth, roles);
        this.symptoms = null;
    }
}
